### PR TITLE
magento/magento2#: Unit test for \Magento\Bundle\Observer\AppendUpsellProductsObserver

### PR DESCRIPTION
**Pull request by atwixfirster**
### Description (*)

PR adds a unit test which cover `\Magento\Bundle\Observer\AppendUpsellProductsObserver` class.

### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

Thank you!

### CODE REVIEW

File: app/code/Magento/Bundle/Test/Unit/Observer/AppendUpsellProductsObserverTest.php [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+                'config' => $this->configMock,
+                'bundleSelection' => $this->bundleSelectionMock,
+            ]
+        );
+    }
+
+    /**
+     * Test observer execute method
+     */
+    public function testExecute()
```
Comment (lbajsarowicz) [inline app/code/Magento/Bundle/Test/Unit/Observer/AppendUpsellProductsObserverTest.php]: Rename to reflect what is given and what are the assertions.

Eg. "testExecuteForXLoadsCollectionOnce"

File: app/code/Magento/Bundle/Test/Unit/Observer/AppendUpsellProductsObserverTest.php [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+     */
+    public function testObserverWithoutBundleIds()
+    {
+        $collectionItems = [
+            1 => 1
+        ];
+        $parentIds = [1];
+        $limit = 2;
+        $productId = 2;
+        $typeId = 'simple';
```
Comment (lbajsarowicz) [inline app/code/Magento/Bundle/Test/Unit/Observer/AppendUpsellProductsObserverTest.php]: You can refer to const from Magento Core

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Great job! Only choose more explanatory name for first test and we are done!

File: app/code/Magento/Bundle/Test/Unit/Observer/AppendUpsellProductsObserverTest.php (hunk 2 above)
Comment (atwixfirster) [reply app/code/Magento/Bundle/Test/Unit/Observer/AppendUpsellProductsObserverTest.php]: **done**

File: app/code/Magento/Bundle/Test/Unit/Observer/AppendUpsellProductsObserverTest.php (hunk 1 above)
Comment (atwixfirster) [reply app/code/Magento/Bundle/Test/Unit/Observer/AppendUpsellProductsObserverTest.php]: sure :)

Thanks, @lbajsarowicz

Comment (atwixfirster) [reply app/code/Magento/Bundle/Test/Unit/Observer/AppendUpsellProductsObserverTest.php]: **done**

**Comment by atwixfirster** [comment]
> Great job! Only choose more explanatory name for first test and we are done!

Suggestions have been implemented.

Could you please review guys?

Many thanks!

**Review by lbajsarowicz — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-6630](https://jira.corp.magento.com/browse/ENGCOM-6630) has been created to process this Pull Request
:eight_spoked_asterisk: @lbajsarowicz, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Alfa** [comment]
@atwixfirster Could you fix failed static tests?
Thanks!

**Comment by atwixfirster** [comment]
> @atwixfirster Could you fix failed static tests?

**fixed**

Thanks for notice, @engcom-Alfa 👍

**Review by dmytro-ch — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @dmytro-ch, thank you for the review.
[ENGCOM-6630](https://jira.corp.magento.com/browse/ENGCOM-6630) has been created to process this Pull Request

**Comment by lbajsarowicz** [comment]
Don't worry about Functional Tests - Fix will be merged soon to `2.4-develop`, then you'll only have to merge latest `2.4-develop` into your feature branch.

**Comment by atwixfirster** [comment]
> Don't worry about Functional Tests - Fix will be merged soon to `2.4-develop`, then you'll only have to merge latest `2.4-develop` into your feature branch.

Thank you for notice, @lbajsarowicz 🙇

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Comment by engcom-Golf** [comment]
failed Functional Tests B2B Not related to the changes in this PR

**Pull request merged**